main.tf

```diff
@@ -824,6 +825,34 @@ data "aws_iam_policy_document" "deny_incorrect_encryption_headers" {
   }
 }
 
+data "aws_iam_policy_document" "deny_incorrect_kms_key_sse" {
+  count = local.create_bucket && var.attach_deny_incorrect_kms_key_sse ? 1 : 0
+
+  statement {
+    sid    = "denyIncorrectKmsKeySse"
+    effect = "Deny"
+
+    actions = [
+      "s3:PutObject"
+    ]
+
+    resources = [
+      "${aws_s3_bucket.this[0].arn}/*"
+    ]
+
+    principals {
```

Comment:
I believe, such verbose `principals` are not needed because they will be like this by default. Please remove it from here and other policies unless you think it is really needed.

Reply:
Removing them give me this explicit error : 
`Error: putting S3 policy: MalformedPolicy: Missing required field Principal`

Reply:
You're right. Thanks for checking out this.